src/LiveComponent/src/Hydration/AbstractDoctrineHydrationExtension.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\LiveComponent\Hydration;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\Persistence\ManagerRegistry;
+use Doctrine\Persistence\ObjectManager;
+
+/**
+ * @author Jean-Paul van der Wegen <info@jpvdw.nl>
+ *
+ * @experimental
+ *
+ * @internal
+ */
+abstract class AbstractDoctrineHydrationExtension
+{
+    /**
+     * @param ManagerRegistry[] $managerRegistries
+     */
+    public function __construct(
+        private iterable $managerRegistries,
+    ) {
+    }
+
+    protected function objectManagerFor(string $class): ?ObjectManager
```

Comment:
Could this abstract class be a trait ? Those methods were private, and i'm not sure we should change their visibility